python-avd/pyavd/_eos_designs/shared_utils/node_config.py

```diff
@@ -57,13 +59,42 @@ def node_config(self: SharedUtilsProtocol) -> EosDesigns._DynamicKeys.DynamicNod
         """
         NodesItem object containing the fully inherited node config.
 
-        Vars are inherited like:
-        <node_type_key>.defaults ->
-            <node_type_key>.node_groups.[<node_group>] ->
-                <node_type_key>.node_groups.[<node_group>].nodes.[<node>] ->
-                    <node_type_key>.nodes.[<node>]
+        This is coming from either node_type_config (like 'l3leaf:') or from the new 'devices'/'device_profiles' models.
+
+        For node_type_config vars are inherited like (first one wins):
+        <node_type_key>.nodes.[<node>] ->
+            <node_type_key>.node_groups.[<node_group>].nodes.[<node>] ->
+                <node_type_key>.node_groups.[<node_group>] ->
+                    <node_type_key>.defaults
+
+        For 'devices' vars are already inherited in self.device_config (first one wins):
+        parent_profiles[name=parent_profile] ->
+            profile[name=profile] ->
+                devices[name=hostname]
         """
-        node_config = self.node_type_config.nodes.get(self.hostname, default=EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem())
+        if self.device_config is not None:
+            # Detect if the device is _also_ defined under the node type model. If so raise an error.
+            if self.node_type_config is not None and (self.hostname in self.node_type_config.nodes or self.node_group_config is not None):
+                msg = f"Found the device '{self.hostname}' under both '{self.node_type_key_data.key}' and 'devices'. Remove the device from one of the models."
+                raise AristaAvdInvalidInputsError(
+                    msg,
+                    host=self.hostname,
+                )
+
+            return self.device_config._cast_as(EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem, ignore_extra_keys=True)
+
+        if self.node_type_config is None:
+            msg = (
+                f"'type' is set to '{self.type}', for which node configs should use the key '{self.node_type_key_data.key}'"
+                f"but '{self.node_type_key_data.key}' was not found. Alternatively use the new 'devices[]' model."
```

Comment:
do we want to suggest the alternative for a PREVIEW model?

Reply:
removed